tests/unit_tests/test_scoring_hpf_scoring.py

```diff
@@ -1,35 +1,152 @@
+        hpf_result = self.hpf_obj.filter_recommendation(normal, 2)
+        self.assertListEqual(filter_result, hpf_result)
+
+    def test_folding_in(self):
+        """Test 2 flows of folding_in() with dummy input set."""
+        input_id_set = {1}
+        final_result = self.hpf_obj.folding_in(input_id_set)
+        recorded_result = [{'cooccurrence_probability': 99.99996940796234,
+                            'package_name': 'org.springframework:spring-core',
+                            'topic_list': []},
+                           {'cooccurrence_probability': 99.99996940796234,
+                            'package_name': 'org.springframework:spring-context',
+                            'topic_list': []},
+                           {'cooccurrence_probability': 99.99996940796234,
+                            'package_name': 'org.springframework:spring-beans',
+                            'topic_list': []},
+                           {'cooccurrence_probability': 99.99996940796234,
+                            'package_name': 'commons-logging:commons-logging',
+                            'topic_list': []},
+                           {'cooccurrence_probability': 99.99996940796234,
+                            'package_name': 'commons-codec:commons-codec',
+                            'topic_list': []}]
+        self.assertListEqual(recorded_result, final_result)
+        input_id_set = {115, 131, 132, 133, 134, 136,
+                        2231, 4599, 4600, 4601, 4602,
+                        4603, 4604, 4605}
+        final_result = self.hpf_obj.folding_in(input_id_set)
+        recorded_result = [{'cooccurrence_probability': 8.3660056968794354,
+                            'package_name': 'org.mongodb:mongo-java-driver',
+                            'topic_list': []},
+                           {'cooccurrence_probability': 8.3989125875990549,
+                            'package_name': 'commons-validator:commons-validator',
+                            'topic_list': []},
+                           {'cooccurrence_probability': 9.3138002424054864,
+                            'package_name': 'org.springframework:spring-jdbc',
+                            'topic_list': []},
+                           {'cooccurrence_probability': 9.3328915508651029,
+                            'package_name': 'org.hectorclient:hector-core',
+                            'topic_list': []},
+                           {'cooccurrence_probability': 9.6898791570348752,
+                            'package_name': 'xom:xom',
+                            'topic_list': []}]
+        self.assertListEqual(recorded_result, final_result)
+
+    def test_predict(self):
+        """Test 2 flows of predict() on dummy input stack."""
+        input_stack = ["io.vertx:vertx-core", "io.vertx:vertx-web"]
+        predict_result = self.hpf_obj.predict(input_stack)
+        predict_list = [{'cooccurrence_probability': 99.99996940796234,
```

Comment:
These numbers seem to be oddly specific. Will they change when we retrain the model?

Reply:
When do retraining from AWS S3 or locally via Minio, then no. Unless we change the data under the `test/test_data` folder, they will not change.